revolution_pi/gpio_pin.go

```diff
@@ -55,18 +101,28 @@ func (pin *gpioPin) Set(ctx context.Context, high bool, extra map[string]interfa
 		val = uint8(1)
 	}
 
-	// disable pwm if enabled
+	if !pin.isOutputPWM() && !pin.isOutputWord() {
+		return fmt.Errorf("cannot set pin state, Pin %s is not a digital output pin", pin.Name)
+	}
+
+	// error if using PWM mode
 	if pin.pwmMode {
-		err := pin.disablePwm()
-		if err != nil {
-			return err
-		}
+		return fmt.Errorf("cannot set pin state, Pin %s is configured as PWM", pin.Name)
```

Comment:
Discussion question: if you've got a PWM pin and someone wants to set it as GPIO output, why can't we just set the duty cycle to 100% or 0% and pretend that's on or off? ("Because that's an ugly hack" is a valid answer!)

Reply:
Would prefer not to do this just to not add even more complexity